<launch>


    <arg name="world_name"/> <!--default="simple_map"/>-->
    <arg name="labels" default="$(find milestone2)/perception/coco.names"/>
    <arg name="initial_map_to_odom" default="-0.5 -0.5 0 0 0 0"/>

    <rosparam file="$(find milestone3)/config/milestone3.yaml" />

    <include file="$(find dd2419_simulation)/launch/aruco.launch"/>

    <node pkg="tf2_ros" type="static_transform_publisher" name="base_to_cam_trans" args="0.01 0 0.02 -1.57 0 -1.57 cf1/base_link cf1/camera_link">
    </node>

    <node pkg="milestone3" type="update_map_to_odom_pose_update_v5.py" name="map_to_odom_publisher">
        <param name="initial_map_to_odom" value="$(arg initial_map_to_odom)"/>
    </node>

    <node pkg="milestone3" type="navgoal_server" name="navgoal_publisher"/>
    <node pkg="milestone3" type="navigation_server.py" name="navigation_server"/>

    <node pkg="milestone3" type="aruco_detection" name="aruco_detection">
    </node>
    <node pkg="milestone3" type="displaymap.py" name="map_markers">
        <param name="world_name" value="$(find dd2419_resources)/worlds_json/$(arg world_name).world.json"/>
    </node>
    <!--<node pkg="milestone2" type="find_object_and_check_it" name="find_n_check">
        <param name="world_name" value="$(find dd2419_resources)/worlds_json/$(arg world_name).world.json"/>
        <param name="labels" value="$(find milestone2)/scripts/$(arg labels)"/>
    </node>-->
    <node pkg="milestone3" type="RRT_ROS.py" name="path_planner">
        <param name="world_name" value="$(find dd2419_resources)/worlds_json/$(arg world_name).world.json"/>
    </node>

</launch>
